mdtk/pytorch_datasets.py

```diff
@@ -1 +1,327 @@
+            returned so, for example, it can be used to convert all data to
+            torch tensors.
+    
+        Returns
+        -------
+        df : pd.DataFrame
+            The pandas DataFrame representing the note data
+        """    
+        self.vocab = vocab
+        self.seq_len = seq_len
+
+        self.in_memory = in_memory
+        self.corpus_lines = corpus_lines
+        self.corpus_path = corpus_path
+        self.encoding = encoding
+        
+        self.transform = transform
+        
+        with open(corpus_path, "r", encoding=encoding) as f:
+            if self.corpus_lines is None and not in_memory:
+                for _ in tqdm.tqdm(f, desc="Counting nr corpus lines"):
+                    self.corpus_lines += 1
+
+            if in_memory:
+                self.lines = [line[:-1].split(",")
+                              for line in tqdm.tqdm(f, desc="Loading Dataset",
+                                                    total=corpus_lines)]
+                self.corpus_lines = len(self.lines)
+
+        if not in_memory:
+            self.file = open(corpus_path, "r", encoding=encoding)
+
+    def __len__(self):
+        return self.corpus_lines
+
+    def __getitem__(self, item):
+        deg_cmd, clean_cmd, deg_num = self.get_corpus_line(item)
+        deg_cmd = self.tokenize_sentence(deg_cmd)
+        deg_cmd = [self.vocab.sos_index] + deg_cmd + [self.vocab.eos_index]
+        clean_cmd = self.tokenize_sentence(clean_cmd)
+        clean_cmd = [self.vocab.sos_index] + clean_cmd + [self.vocab.eos_index]
+        deg_num = int(deg_num)
+        
+        deg_cmd = deg_cmd[:self.seq_len]
+        deg_cmd += [self.vocab.pad_index for _ in 
+                    range(self.seq_len - len(deg_cmd))]
+        clean_cmd = clean_cmd[:self.seq_len]
+        clean_cmd += [self.vocab.pad_index for _ in 
+                      range(self.seq_len - len(clean_cmd))]
+        
+        output = {"deg_commands": deg_cmd,
+                  "clean_commands": clean_cmd,
+                  "deg_label": deg_num}
+        
+        # TODO: implement transform as in https://pytorch.org/tutorials/beginner/data_loading_tutorial.html
+        if self.transform is not None:
+            output = self.transform(output)
+        return output 
+
+    def tokenize_sentence(self, sentence):
+        tokens = sentence.split()
+        for ii, token in enumerate(tokens):
+            tokens[ii] = self.vocab.stoi.get(token, self.vocab.unk_index)
+        return tokens
+
+    def get_corpus_line(self, item):
+        if self.in_memory:
+            deg_cmd, clean_cmd, deg_num = self.lines[item]
+            return deg_cmd, clean_cmd, deg_num
+        else:
+            line = self.file.__next__()
+            if line is None:
+                self.file.close()
+                self.file = open(self.corpus_path, "r", encoding=self.encoding)
+                line = self.file.__next__()
```

Comment:
This won't work with shuffling, correct?

Reply:
Ah, you mentioned that in the docstring!

Reply:
haha! Yes! It's the functionality i've adapted from https://github.com/codertimo/BERT-pytorch. It's not ideal but at least something is in place that we can change later (I'm sure there must be a way of reading a specific line from a file (I think `sed` can do it for instance)